mplx_qc.py

```diff
@@ -103,89 +106,114 @@ def read_input(input_file):
     for row in row_iter:
         merged_cram = Generic()
         for column_name, cell in zip(column_names, row):
-            if column_name in ['json_path', 'cram_path']:
+            if column_name in ['merge_id', 'json_path', 'cram_path']:
                 value = cell.value
                 setattr(merged_cram, column_name, value)
         merged_crams.append(merged_cram)
     return merged_crams
 
 
-def process_crams(cram_paths):
-    """Read cram_paths, parse RGs List and then
+def compare_read_groups(cram_path, json_path):
+    """Compare a set of CRAM RG barcodes & samples to JSON barcodes & samples
+    for one merged CRAM, returning most severe error code.
+
+    Expectations:
+        set(cram_rg_barcodes) == set(json_rg_barcodes)
+        len(set(cram_rg_barcodes)) == len(cram_rg_barcodes)
+        len(set(json_rg_barcodes)) == len(json_rg_barcodes)
+        len(set(cram_rg_samples)) == len(set(json_rg_samples)) == 1
+        set(cram_rg_samples) == set(json_rg_samples)
+    """
+    cram_rg_barcodes, cram_rg_samples = process_cram(cram_path)
+    json_rg_barcodes, json_rg_samples = process_json(json_path)
+    logger.info('found %s cram_rg_samples, %s json_rg_samples',
```

Comment:
Oops! We're logging the number of barcodes found, but the text talks about samples, which will always be 1 in healthy merges.